app/src/test/java/com/arielwang/workoutlogger/exercisecard/ExerciseCardViewModelTest.kt

```diff
@@ -0,0 +1,160 @@
+            TODO("Not yet implemented")
+        }
+
+    }
+
+    @Test
+    fun `When topbar back button is clicked, go back to previous page`() {
+        runTest {
+            val viewModel = generateViewModel()
+            val goBackToPreviousPageButton = ExerciseCardView.Action.GoBackToPreviousPage
+
+            viewModel.onUiAction(goBackToPreviousPageButton)
+
+            assertEquals(
+                FakeNavigationBehaviour.NavigateUp,
+                fakeNavigator.awaitNextScreen()
+            )
+        }
+    }
+
+    @Test
+    fun `When the value of TextFiled in nameSection changes, update ExerciseCardView State name`() {
+        runTest {
+            val viewModel = generateViewModel()
+            val onTextChange = ExerciseCardView.Action.OnTextFieldValueChangeNameText("Core")
+
+            viewModel.onUiAction(onTextChange)
+
+            viewModel.uiState.test {
+                assertEquals(
+                    ExerciseCardView.State(
+                        name = "Core"
+                    ),
+                    awaitItem()
+                )
+            }
+        }
+    }
+
+    @Test
+    fun `When the value of TextFiled in notesSection changes, update ExerciseCardView State notes`() {
+        runTest {
+            val viewModel = generateViewModel()
+            val onTextChange = ExerciseCardView.Action.OnTextFieldValueChangeNoteText("Repeat twice")
+
+            viewModel.onUiAction(onTextChange)
+
+            viewModel.uiState.test {
+                assertEquals(
+                    ExerciseCardView.State(
+                        notes = "Repeat twice"
+                    ),
+                    awaitItem()
+                )
+            }
+        }
+    }
+
+    @Test
+    fun `When clear button in nameSection is clicked, update ExerciseCardView State name`() {
+        runTest {
+            val viewModel = generateViewModel()
+            val onClearButtonClicked = ExerciseCardView.Action.ClearName
+
+            viewModel.onUiAction(onClearButtonClicked)
+
+            viewModel.uiState.test {
+                assertEquals(
+                    ExerciseCardView.State(
+                        name = ""
+                    ),
+                    awaitItem()
+                )
+            }
+        }
+    }
+
+    @Test
+    fun `When clear button in notesSection is clicked, update ExerciseCardView State notes`() {
+        runTest {
+            val viewModel = generateViewModel()
+            val onClearButtonClicked = ExerciseCardView.Action.ClearNote
+
+            viewModel.onUiAction(onClearButtonClicked)
+
+            viewModel.uiState.test {
+                assertEquals(
+                    ExerciseCardView.State(
+                        notes = ""
+                    ),
+                    awaitItem()
+                )
+            }
+        }
+    }
+
+    // when saved button is clicked, insert exercise card data into database and go back to Exercise screen
```

Comment:
We can simply use this as the function name

Reply:
Thank you.